[WIP needs review] Editing README

sishida opened the pull request:
Signed-off-by: Sarah Ishida <sishida@us.ibm.com>

I am creating this pull request for https://github.com/eclipse/codewind/issues/596.

Some thoughts:
- I removed the feature list that was originally included because it is a duplicate list from the Codewind.dev site, and it's extra work to keep the list up to date in both places.
- If we want to move the "Developing" and "Building Codewind from the source" sections because they don't relate to a new user installing Codewind for the first time, perhaps we could add them to the Contributing.md file or a wiki page within the `codewind-vscode` repo?

rtaniwa commented on README.md:
```diff
@@ -4,7 +4,20 @@
 
 # Codewind for VS Code
-Create and develop cloud-native, containerized web applications from VS Code.
+Codewind for VS Code provides tools to help build cloud-native, containerized web applications from VS Code, regardless of which IDE or language you use. Rapidly create an application from a template project and launch, update, test, and debug in Docker containers on the desktop.
```
I would say "containerized applications" as they aren't always just web.  Also because they would be viewing this from the VS Code marketplace, we may be able to eliminate the "IDE or" and just say regardless of which language you use.

tetchel replied:
it doesn't make sense to include "regardless of which IDE you use" in the VS Code readme

tetchel commented on README.md:
```diff
@@ -4,7 +4,20 @@
+## Features
+- Create new containerized projects or add existing ones.
+- View your containerized projects, including the application and build status.
+- Debug Microprofile/Java EE and Spring applications.
```
and nodejs

tetchel commented on README.md:
```diff
@@ -4,7 +4,20 @@
+- Create new containerized projects or add existing ones.
+- View your containerized projects, including the application and build status.
+- Debug Microprofile/Java EE and Spring applications.
+- Set up a debug session in a Chromium-based browser for Node.js applications.
```
Not true for vs code

tetchel commented on README.md:
```diff
@@ -4,7 +4,20 @@
+- View your containerized projects, including the application and build status.
+- Debug Microprofile/Java EE and Spring applications.
+- Set up a debug session in a Chromium-based browser for Node.js applications.
+- View application and build logs in the Console view.
```
in vs code it's the Output view, not Console

tetchel commented on README.md:
```diff
@@ -4,7 +4,20 @@
+- Debug Microprofile/Java EE and Spring applications.
+- Set up a debug session in a Chromium-based browser for Node.js applications.
+- View application and build logs in the Console view.
+- Integrate validation errors into the Markers view.
```
"Markers" -> "Problems"

rtaniwa replied:
"See validation errors in the Markers view" seems more accurate. "Integrate" makes it sound like developers themselves would have some ability to integrate validation errors.

tetchel commented:
putting aside the differences from Eclipse I pointed out already, i'm still not sure that listing a bunch of Codewind terms/features really solves rob's original problem of "why should I use codewind"

sishida commented:
Before I make more edits, I wanted to acknowledge the comment from @tetchel about user understanding--I think it's a good point. Instead of a list of features, would it be better to have a list of possible benefits? Or make the feature list broader/more generic so that more users would be able to understand how Codewind can help them?

Maybe something along the lines of what we have on Codewind.dev?
- Get started quickly with templates or samples.
- Pull in existing applications and get them ready for the cloud.
- Enjoy quick, iterative development and fast feedback directly in your IDE.
- Learn as you go.

Do you prefer something more like this, @rtaniwa?
Thanks, team!

rtaniwa commented:
Hey @sishida. Ideally it would have both value props, and a feature list. I have been having numerous conversations about our entry points, and how it isn't always clear to those reading it what benefits they get if they used Codewind. I think this can be applied to our landing pages and our marketplace entries. I think the low hanging fruit was bringing over some of what we had on the Eclipse marketplace to the VS Code marketplace (because it had more depth), but the full answer is we should revisit the copy and content and make sure what we have there sells the benefits of Codewind to a developer.

sishida commented:
@rtaniwa and @tetchel, Thank you for your feedback!

In regards to the discussion about value props and the feature list, I have added a "Why use Codewind?" section (based on info from the Codewind website home page) in addition to retaining and updating the existing feature list according to your feedback.

Please let me know if you'd like further changes. Thank you!

tetchel commented on README.md:
```diff
@@ -4,7 +4,27 @@
+Codewind for VS Code provides tools to help build cloud-native, containerized applications from VS Code, regardless of which language you use. Rapidly create an application from a template project and launch, update, test, and debug in Docker containers on the desktop.
+
+## Why use Codewind?
+- Work on your desktop then run and debug on Kubernetes.
```
"debug on Kubernetes" is misleading because we don't support debugging while running on Kube yet, that is a ways away still.

We _do_ support debugging applications in Docker containers, which is worth mentioning (it is mentioned in the Features list below).
We _will_ support _deploying_ from desktop to Kubernetes, but that has not hit production yet.

tetchel commented on README.md:
```diff
@@ -4,7 +4,27 @@
 
 # Codewind for VS Code
-Create and develop cloud-native, containerized web applications from VS Code.
+Codewind for VS Code provides tools to help build cloud-native, containerized applications from VS Code, regardless of which language you use. Rapidly create an application from a template project and launch, update, test, and debug in Docker containers on the desktop.
```
maybe instead of 

> regardless of which language you use

we could broaden that to

> regardless of which technologies or language you use.
or 
> regardless of which runtime or language you use.
or
> regardless of which application stack you use.

tetchel commented:
:+1: definitely a step in the right direction

sishida commented:
I have added a new commit to address the latest comments from @tetchel.
- I removed the line about debugging on Kubernetes.
- Line 7 now reads "runtime or language" because I thought those terms were the least ambiguous. 👍

tetchel commented:
@rtaniwa can you review?

rtaniwa commented on README.md:
```diff
@@ -4,7 +4,26 @@
+Codewind for VS Code provides tools to help build cloud-native, containerized applications from VS Code, regardless of which runtime or language you use. Rapidly create an application from a template project and launch, update, test, and debug in Docker containers on the desktop.
+
+## Why use Codewind?
+- Get started quickly with templates or samples.
```
Get started quickly with templates or samples that allow you to quickly create and deploy applications running in containers

rtaniwa commented on README.md:
```diff
@@ -4,7 +4,26 @@
+
+## Why use Codewind?
+- Get started quickly with templates or samples.
+- Use the tools, runtimes, and languages that you're already familiar with.
```
Supports all the popular runtimes, allowing you to create containerized applications in languages your are already familiar with

rtaniwa commented on README.md:
```diff
@@ -4,7 +4,26 @@
+## Why use Codewind?
+- Get started quickly with templates or samples.
+- Use the tools, runtimes, and languages that you're already familiar with.
+- Pull in existing applications and get them ready for the cloud.
```
"Pull in your existing applications, and let Codewind help you get them cloud ready"

I think we need to be a bit careful here, because it isn't always easy for us in current state to detect what stuff may or may not be missing to enable the application to run in a container. And this can vary across the different application types we support (ie. Appsody).

rtaniwa commented on README.md:
```diff
@@ -4,7 +4,26 @@
+- Get started quickly with templates or samples.
+- Use the tools, runtimes, and languages that you're already familiar with.
+- Pull in existing applications and get them ready for the cloud.
+- Enjoy quick, iterative development and fast feedback.
```
Our advanced inner loop ensures that we manage your application updates efficiently, so that your code changes are reflected in your containerized application almost instantaneously.

rtaniwa commented on README.md:
```diff
@@ -4,7 +4,26 @@
+- Use the tools, runtimes, and languages that you're already familiar with.
+- Pull in existing applications and get them ready for the cloud.
+- Enjoy quick, iterative development and fast feedback.
+- Learn as you go.
```
Learn as you go feels a bit weak. Maybe we can say "Development flow within the IDE feels  the same as traditional application development - we allow you to develop in containers without it feeling like you are developing in containers"

or something along those lines...maybe needs shortening?

rtaniwa commented on README.md:
```diff
@@ -4,7 +4,26 @@
+- Create new containerized projects or add existing ones.
+- View your containerized projects, including the application and build status.
+- Debug Microprofile/Java EE, Node.js, and Spring applications.
+- View application and build logs in the Output view.
```
Maybe "Access application, build, and container logs in the Output view"

rtaniwa commented on README.md:
```diff
@@ -4,7 +4,26 @@
+- See validation errors in the Problems view.
+- Open a shell session into an application container.
+- Toggle the project auto build setting and manually initiate project builds.
+- Open your application or the application monitor in a browser.
```
Maybe instead of Open we say "Launch"?

sishida replied:
Our word usage page says, "Do not use unless it is part of the terminology of the product that you are documenting. Use "start" or "open". I changed it to "start" if that's OK.

rtaniwa commented:
@sishida additional feedback provided. Apologies, I didn't see the request for review earlier in the week.

sishida commented:
@rtaniwa Thank you for your feedback. I have added changes to my latest commit. If the changes look OK, could you please add your approval? Thank you! 😄
